openshift/default-catalog-consistency/pkg/check/filesystem.go

```diff
@@ -38,3 +41,76 @@ func FilesystemHasDirectories(paths ...string) FilesystemCheck {
 		},
 	}
 }
+
+// FilesystemHasExecutables checks whether the specified paths refer to existing, accessible, and executable file.
+// It handles both regular files and symbolic links:
+//   - For regular files: checks that the file exists and is executable.
+//   - For symlinks: follows the link and checks that the target is an executable regular file.
+//
+// Any issues (missing file, broken symlink, non-executable file, etc.) will be reported as errors.
+func FilesystemHasExecutables(root string, paths ...string) FilesystemCheck {
```

Comment:
Maybe I'm missing something, but it looks like we are calling this function with `bin/opm` as `root` with no `paths`.

But the function doesn't really do anything unless there are paths to iterate.

Reply:
Sorry, with your ocmment I realized that I sent the wrong version of the solution fixed.
We are passing in the func, I change the interface.